### DISCUSSION TITLE
Lots of cache misses

### DISCUSSION THREAD

**Discussion by timschumi** [ccache/ccache#166] [Support]
Since I started mounting my source directory with "mount -o bind" to my computer's /tmp folder, I only got cache misses on first compile. When recompiling, everything is fine. Before I started mounting, I got a miss rate of about 1 to 2500. Is that due to the changed path or am I missing something?

EDIT: I also changed the output directory. At first I tried compiling from the original directory, which also lead to cache misses. Then I set back the Output to the original directory too and there are no cache misses so far

EDIT2: It looks like the output directory needs to (relatively) be the same when ccache should recognize it correctly. I'll look through the Makefiles and see what was altered when changing the path

**Comment by afbjorklund** [comment]
If you are building from a `mktemp` location, you will need to look into `base_dir`.

https://github.com/ccache/ccache/blob/master/MANUAL.txt#L779

Normally output doesn't matter, but maybe you have it in your include path or something.

**Comment by timschumi** [comment]
It seems like the relative path to the output always needs to be the same. I couldn't find anything in the Makefiles though, they are too obfuscated to see anything.

I found another solution, namely linking the output directory itself, not the source directory.

I'll close this now. Thanks for the help.
